modules/raft/src/integrationTest/java/org/apache/ignite/raft/server/ItJraftDestructorTest.java

```diff
@@ -0,0 +1,145 @@
+ * limitations under the License.
+ */
+
+package org.apache.ignite.raft.server;
+
+import static org.apache.ignite.internal.testframework.IgniteTestUtils.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.stream.Stream;
+import org.apache.ignite.internal.cluster.management.CmgGroupId;
+import org.apache.ignite.internal.lang.IgniteInternalException;
+import org.apache.ignite.internal.metastorage.server.raft.MetastorageGroupId;
+import org.apache.ignite.internal.raft.Peer;
+import org.apache.ignite.internal.raft.RaftNodeId;
+import org.apache.ignite.internal.raft.server.RaftGroupOptions;
+import org.apache.ignite.internal.raft.server.impl.JraftServerImpl;
+import org.apache.ignite.internal.raft.storage.LogStorageFactory;
+import org.apache.ignite.internal.raft.storage.impl.LogStorageException;
+import org.apache.ignite.internal.replicator.ReplicationGroupId;
+import org.apache.ignite.internal.replicator.TablePartitionId;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/** Tests that check that failed storage destruction is finished after server restart. */
+public class ItJraftDestructorTest extends JraftAbstractTest {
+
+    private static Stream<ReplicationGroupId> groupIdProvider() {
+        return Stream.of(
+                new TablePartitionId(0, 0),
+                CmgGroupId.INSTANCE,
+                MetastorageGroupId.INSTANCE
+        );
+    }
+
+    private JraftServerImpl server;
+    private Path serverDataPath;
+    private Peer peer;
+
+    @BeforeEach
+    void setUp() {
+        server = startServer();
+
+        serverDataPath = serverWorkingDirs.get(0).basePath();
+
+        String localNodeName = server.clusterService().topologyService().localMember().name();
+
+        peer = Objects.requireNonNull(initialMembersConf.peer(localNodeName));
+
+        LogStorageFactory logStorageFactory = logStorageFactories.get(0);
+
+        doThrow(LogStorageException.class).doCallRealMethod().when(logStorageFactory).destroyLogStorage(anyString());
+    }
+
+    @ParameterizedTest
+    @MethodSource("groupIdProvider")
+    void testFinishStorageDestructionAfterRestart(ReplicationGroupId replicationGroupId) throws Exception {
+        RaftNodeId nodeId = new RaftNodeId(replicationGroupId, peer);
+
+        destroyStorageFails(server, nodeId, serverDataPath, false);
+
+        Path nodeDataPath = createNodeDataDirectory(nodeId);
+
+        restartServer();
+
+        assertFalse(Files.exists(nodeDataPath));
```

Comment:
Let's assert that this directory exists before an attempt to destroy storages - just to make sure we are looking at the correct directory